pkg/runner/runner.go

```diff
@@ -13,28 +14,39 @@ import (
 	"net/http"
 	"time"
 
+	"github.com/google/uuid"
 	_ "github.com/jackc/pgx/v4/stdlib"
+	"go.opentelemetry.io/collector/model/otlpgrpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
 	"github.com/jaegertracing/jaeger/plugin/storage/grpc/shared"
 	"github.com/thanos-io/thanos/pkg/store/storepb"
+
 	"github.com/timescale/promscale/pkg/api"
 	"github.com/timescale/promscale/pkg/jaeger/query"
 	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/telemetry"
 	"github.com/timescale/promscale/pkg/thanos"
 	"github.com/timescale/promscale/pkg/util"
 	tput "github.com/timescale/promscale/pkg/util/throughput"
 	"github.com/timescale/promscale/pkg/version"
-	"go.opentelemetry.io/collector/model/otlpgrpc"
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials"
 )
 
 const promLivenessCheck = time.Second
 
 var (
-	elector      *util.Elector
+	elector     *util.Elector
+	PromscaleID uuid.UUID
+
 	startupError = fmt.Errorf("startup error")
 )
 
+func init() {
+	// PromscaleID must always be generated on start, so that it remains constant throughout the lifecycle.
+	PromscaleID = uuid.New()
```

Comment:
This seems to imply that on every instance restart you get a new PromscaleID. Shouldn't we keep and re-use the same UUID in the database? So on first/initial run we will populate it in the DB and on restart we look it up and load into memory

Reply:
The `promscale_instance_information` table needs a UUID per Promscale instance, to identify them individually, hence we need it per promscale. Reason: when the stats (say num_samples_ingested) are related to a X promscale updates, promscale should know which row it should update and not touch the rows of other promscales that are running.